packages/flame_texturepacker/lib/src/texture_packer_sprite.dart

```diff
@@ -97,42 +150,67 @@ class TexturePackerSprite extends Sprite {
     Anchor anchor = Anchor.topLeft,
     Paint? overridePaint,
   }) {
-    if (!rotate) {
-      return super.render(
-        canvas,
-        position: position,
-        size: size,
-        anchor: anchor,
-        overridePaint: overridePaint,
-      );
-    }
     if (position != null) {
       _tmpRenderPosition.setFrom(position);
     } else {
       _tmpRenderPosition.setZero();
     }
 
-    // If the sprite is rotated on the sprite sheet un-rotate it
-    // and adjust the size
-    final auxAnchor = rotate ? Anchor.bottomLeft : anchor;
+    // Get sprite size
+    _tmpRenderSize.setFrom(size ?? srcSize);
+
+    // Calculate topLeft position
+    _tmpRenderPosition.setValues(
+      _tmpRenderPosition.x - (anchor.x * _tmpRenderSize.x),
+      _tmpRenderPosition.y - (anchor.y * _tmpRenderSize.y),
+    );
+
+    // Calculate multiplier value
+    _tmpRenderScale
+      ..setFrom(_tmpRenderSize)
+      ..divide(originalSize);
+
+    // Calculate image size rendered based on packedSize
+    _tmpRenderImageSize
+      ..setFrom(_packedSize)
+      ..multiply(_tmpRenderScale);
 
-    final tempSize = size ?? srcSize;
-    final tempWidth = rotate ? tempSize.y : tempSize.x;
-    final tempHeight = rotate ? tempSize.x : tempSize.y;
+    // Calculate image rendered offset from topLeft position
+    _tmpRenderOffset
+      ..setFrom(offset)
+      ..multiply(_tmpRenderScale)
+      ..add(_tmpRenderPosition);
 
-    _tmpRenderSize.setValues(tempWidth, tempHeight);
+    if (!rotate) {
+      // Calculate and render for non-rotated image, must call function render
+      // from super class with anchor = Anchor.topLeft, because we already
+      // calculated size and position based on Anchor.topLeft to rendered by
+      // super class function.
+      _tmpRenderSize.setFrom(_tmpRenderImageSize);
+      _tmpRenderPosition.setFrom(_tmpRenderOffset);
+      return super.render(
+        canvas,
+        position: _tmpRenderPosition,
+        size: _tmpRenderSize,
+        overridePaint: overridePaint,
+      );
+    }
 
+    // Calculate and render for non-rotated image, must call function render
```

Comment:
```suggestion
    // Calculate and render for rotated image, must call function render
```

Reply:
Sorry my bad..
With update it at next commit.

Reply:
done